packages/server/lib/crons/delete/deleteProviderConfigData.ts

```diff
@@ -15,6 +16,20 @@ export async function deleteProviderConfigData(providerConfig: IntegrationConfig
     const { logger, deadline, limit } = opts;
     logger.info('Deleting provider config...', providerConfig.id, providerConfig.unique_key);
 
+    await batchDelete({
+        ...opts,
+        name: 'connections < providerConfigs',
+        deleteFn: async () => {
+            const connections = await db.knex.from<DBConnection>('_nango_connections').where({ config_id: providerConfig.id! }).limit(opts.limit);
+
+            for (const connection of connections) {
+                await deleteConnectionData(connection, opts);
+            }
+
+            return connections.length;
+        }
+    });
```

Comment:
Can you just delete connections after the next block, otherwise the FK will still try to delete N syncs